### DISCUSSION TITLE
Allow providing pre-compiled `main.jsbundle` (CI optimization)

### DISCUSSION THREAD

**Discussion by andreialecu** [react-native-community/discussions-and-proposals#376] [Ideas]
### Introduction

Since MacOS compute time is generally the most expensive on CI platforms, it seems that it would be possible to reduce the amount of time taken during `xcode` compilation by skipping the `react-native bundle` compilation step.

For context, I wanted to create a workflow to run `react-native bundle` on Linux VMs and then feed the pre-compiled assets to a MacOS build step further down the pipeline, but ran into issues.

CircleCI specifically mentions this possibility on their RN landing page: https://circleci.com/circleci-react/

> Run Devtools in Linux
> Build macOS and run devtools in Linux jobs to increase overall build throughput.

Additionally, depending on the complexity of the pipeline, the bundle could be reused or even cached in a CI workflow, greatly reducing overall processing time.

### Details

There is a `SKIP_BUNDLING` env variable that is checked here: https://github.com/facebook/react-native/blob/master/scripts/react-native-xcode.sh#L30

Setting `SKIP_BUNDLING=true` as an environment variable does work to skip the bundling step. 

However, I found it hard to figure out where to actually place the assets, because the destination folder seems to be `DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH`, which is largely unknown before running `XCode`.

### Discussion points

Seems like it might be a good idea to add a feature of reusing pre-built bundles. 

For example, an env variable like `export RN_BUNDLE_PREBUILD_DIR=./dist/ios` could be checked and if present, the build step would simply copy the contents of that directory over to its expected `$DEST` directory. It would be great if this worked for both Android and iOS. For iOS it should be a very simple change to the above `.sh` script. I haven't checked Android yet.

Any thoughts?

**Comment by retyui** [comment]
@andreialecu 
Usually, compiling js bundle it a small piece of the build process a lot of time `xcode` takes to build a native code 


You can try the next way to reduce financial expenses: https://dev.to/retyui/react-native-how-speed-up-ios-build-4x-using-cache-pods-597c

**Comment by andreialecu** [comment]
@retyui You may want to also take a look at https://github.com/Subito-it/PodBuilder/issues/10 for a less complicated solution for prebuilding pods that doesn't necessarily require `fastlane`. 

I haven't been able to use other solutions for prebuilding in real-life projects (such as the ones mentioned in your article) due to various issues which are detailed in the thread above.

**Comment by retyui** [comment]
I read that [Wix engineer](https://omribruchim.medium.com/react-native-at-wix-the-architecture-ii-deep-dive-9cfcb3c2822c) use "expo" way to run js code on once precompile app
